quill-core/src/main/scala/io/getquill/quotation/Quotation.scala

```diff
@@ -54,4 +74,32 @@ trait Quotation extends Parsing with Liftables with Unliftables {
       annotation <- method.annotations.headOption
       astTree <- annotation.tree.children.lastOption
     } yield (astTree)
+
+  private class FreeVarTraverser extends Traverser {
```

Comment:
Would it be possible to parse the tree using the normal parser without free vars, collect the free vars using the current `FreeVariables` function and then parse again? I'd like to avoid using more macro apis, they are in general very fragile.